### ISSUE TITLE
Feature Request: Checkout accessories to assets

### ISSUE THREAD

**Issue by PetzJohannes** [snipe/snipe-it#6865]
**Server (please complete the following information):**
 - snipe-it version: 5.X

**Is your feature request related to a problem? Please describe.**
We have some keyboards and mice that are located on a shared desk. If I would checkout them to a special user, it would be wrong because these are not the accessories personally added to him.

**Describe the solution you'd like**
I would like to checkout accessories to users and assets.

**Describe alternatives you've considered**
If I would change the keyboards and mice type to components, it would be possible to check them out properly. But components cannot be linked to users. So we have another problem.

**Additional context**
We could add keyboards and mice as components and accessories, but this would be a bad set up.
If you like this feature and would accept a feature request for this, I could help to implement this.

**Comment by ajsy** [comment]
+1

**Comment by Nesteph** [comment]
Could this be extended to follow the same checkout system that Assets have (checkout to User / Asset / Location) and also apply this to components?

**Comment by iainhallam** [comment]
Yes, still relevant.

**Comment by iainhallam** [comment]
Running out of ways to tell this bot that issues are still relevant only two months later...

**Comment by bra1ncramp** [comment]
We really need this. We have kits that contain Assets and Accessories. Without being able to check out the accessory to the asset, we cannot account for them correctly.

**Comment by iainhallam** [comment]
Once again...

**Comment by bra1ncramp** [comment]
Yes, unless this feature has been implemented, it is still unresolved. Please do not close this.On Feb 13, 2020 9:32 PM, "stale[bot]" <notifications@github.com> wrote:Is this still relevant? We haven't heard from anyone in a bit. If so, please comment with any updates or additional detail.
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Don't take it personally, we just need to keep a handle on things. Thank you for your contributions!

**Comment by bra1ncramp** [comment]
Yes, this had not been implemented yetOn Apr 14, 2020 1:54 AM, "stale[bot]" <notifications@github.com> wrote:
Is this still relevant? We haven't heard from anyone in a bit. If so, please comment with any updates or additional detail.
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Don't take it personally, we just need to keep a handle on things. Thank you for your contributions!

**Comment by bra1ncramp** [comment]
Unless the feature has been implemented, it is still relevantOn Jun 13, 2020 8:17 AM, "stale[bot]" <notifications@github.com> wrote:
Is this still relevant? We haven't heard from anyone in a bit. If so, please comment with any updates or additional detail.
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Don't take it personally, we just need to keep a handle on things. Thank you for your contributions!

**Comment by iainhallam** [comment]
Relevant

**Comment by bra1ncramp** [comment]
This isn't stale until implemented

**Comment by germanangst** [comment]
Definetely a nice feature. We need to reorganize our assets and would go for accesories. Do you have a database table cheat? I see, that assets and accesories are different tables.

**Comment by AquaDocs** [comment]
This would be an awesome addition: checkout accessories to asset/user/location.

**Comment by Weretoad** [comment]
Throwing my hat in the ring, we would love to see this feature.

**Comment by MrMontesa** [comment]
Same here.
We would like to checkout an accessories to locations.
Our use-case: We want to add desks as accessories. Those belong to a location and should be checked out to those. But same applies to keyboards in out meeting rooms. These accessories should be checked out to a location as well.
Thanks

**Comment by snipe** [comment]
As I've mentioned elsewhere, this is not on the short-term roadmap. We don't encourage people to check things out to non-sentient objects. At the end of the day, if that accessory is damaged or missing, an actual human has to be held responsible for it.

**Comment by pc-h353** [comment]
> As I've mentioned elsewhere, this is not on the short-term roadmap. We don't encourage people to check things out to non-sentient objects. At the end of the day, if that accessory is damaged or missing, an actual human has to be held responsible for it.

I do not get your train of thoughts, but maybe it is just me. Anyway..
With all due respect - why you should be the one, who decides? Don't you think it is always an Admin, who makes that decision? I mean, give the customers the possibility to decide at least.

**Comment by Weretoad** [comment]
> 
> 
> As I've mentioned elsewhere, this is not on the short-term roadmap. We don't encourage people to check things out to non-sentient objects. At the end of the day, if that accessory is damaged or missing, an actual human has to be held responsible for it.

Then why are we able to check out assets to locations? If a location can have an asset, an asset should be able to have an accessory.

**Comment by kale1d0code** [comment]
> As I've mentioned elsewhere, this is not on the short-term roadmap. We don't encourage people to check things out to non-sentient objects. At the end of the day, if that accessory is damaged or missing, an actual human has to be held responsible for it.

I would say that the person who checkouts the asset that the accessory would be checked out to, would be responsible for the asset, and the accessory, as well as anything else checked out to the asset.

My company does this currently with laptop chargers.

**Comment by adagioajanes** [comment]
@kale1d0code I think then it would make more sense to treat these items as components rather than accessories. That is just my opinion. I'd love to hear though if you think that wouldn't work.

**Comment by kiriaki** [comment]
@snipe There are cases where we don't necessarily want to keep someone accountable for all assets, accessories or consumables. Rather we want to keep track of their location. This is the case with shared labs or conference rooms. Please, consider adding this feature in your software. It would definitely be used by some users.

In the meantime, I will try the workaround proposed by @adagioajanes.

**Comment by alexanderpas** [comment]
> As I've mentioned elsewhere, this is not on the short-term roadmap. We don't encourage people to check things out to non-sentient objects. At the end of the day, if that accessory is damaged or missing, an actual human has to be held responsible for it.

The person responsible would be the manager for that location. (with the manager changing over time, but the location not.)

@snipe A primary usecase for checkout to User / Asset / Location for accessories would be for things like laptop chargers and docking stations, which can be permanently deployed at a location such as a meeting room, so people don't have to bring and connect their own docking station to the meeting room, but at the same time people can have them checked out personally, such as those in a WFH situation.

You want to know when you are running low on the amount of chargers or docking station as well as see if you need to buy additional items before setting up a new meeting room, taking into account the amount already in use by both WFH users as well as permanently deployed at locations such as other meeting rooms etc.

**Comment by MSWork79** [comment]
Think this is a duplicate of #5140

**Comment by snipe** [comment]
This was done a while back.
